core/base/inc/TDirectory.h

```diff
@@ -33,66 +33,69 @@ class TFile;
+      TContext   *fNext{nullptr};                   //! Pointer to the next TContext in the implied list of context pointing to fPrevious.
+
+      TContext(TContext&) = delete;
+      TContext& operator=(TContext&) = delete;
+
       void CdNull();
       friend class TDirectory;
    public:
-      TContext(TDirectory *previous, TDirectory *newCurrent)
-         : fDirectory(previous), fActiveDestructor(false), fDirectoryWait(false), fPrevious(0), fNext(0)
+      TContext(TDirectory *previous, TDirectory *newCurrent) : fDirectory(previous)
       {
          // Store the current directory so we can restore it
          // later and cd to the new directory.
-         if ( fDirectory ) (*fDirectory).RegisterContext(this);
-         if ( newCurrent ) newCurrent->cd();
-         else CdNull();
+         if (fDirectory)
+            (*fDirectory).RegisterContext(this);
+         if (newCurrent)
+            newCurrent->cd();
+         else
+            CdNull();
       }
-      TContext()
-         : fDirectory(TDirectory::CurrentDirectory()), fActiveDestructor(false), fDirectoryWait(false), fPrevious(0),
-           fNext(0)
+      TContext() : fDirectory(TDirectory::CurrentDirectory())
       {
          // Store the current directory so we can restore it
          // later and cd to the new directory.
-         if ( fDirectory ) (*fDirectory).RegisterContext(this);
+         if (fDirectory)
+            (*fDirectory).RegisterContext(this);
       }
-      TContext(TDirectory *newCurrent)
-         : fDirectory(TDirectory::CurrentDirectory()), fActiveDestructor(false), fDirectoryWait(false), fPrevious(0),
-           fNext(0)
+      TContext(TDirectory *newCurrent) : fDirectory(TDirectory::CurrentDirectory())
       {
          // Store the current directory so we can restore it
          // later and cd to the new directory.
-         if ( fDirectory ) (*fDirectory).RegisterContext(this);
-         if ( newCurrent ) newCurrent->cd();
-         else CdNull();
+         if (fDirectory)
+            (*fDirectory).RegisterContext(this);
+         if (newCurrent)
+            newCurrent->cd();
+         else
+            CdNull();
       }
       ~TContext();
    };
 
 protected:
 
-   TObject      *fMother;          //pointer to mother of the directory
-   TList        *fList;            //List of objects in memory
-   TUUID         fUUID;            //Unique identifier
-   TString       fPathBuffer;      //!Buffer for GetPath() function
-   TContext     *fContext;         //!Pointer to a list of TContext object pointing to this TDirectory
+   TObject         *fMother{nullptr};   // pointer to mother of the directory
+   TList           *fList{nullptr};     // List of objects in memory
+   TUUID            fUUID;              // Unique identifier
+   mutable TString  fPathBuffer;        //! Buffer for GetPath() function
```

Comment:
Because of `const char *TDirectory::GetPath() const` method.
It uses `fPathBuffer` as container. Either one uses const_cast or simply declares `fPathBuffer` mutable.